### PR TITLE
RPC endpoints for excluding tracing failures

### PR DESCRIPTION
**Pull request by jewei1997**
## Describe your changes and provide context

- [x] add 3 new endpoints
- [x] add unit tests
- [x] name them all *ExcludeTracingFail for clarity
- [x] write a README about old synthetic endpoints + these new endpoints
- [x] test endpoints on RPC

A panic tx is a tx which fails tracing. This can be due to several issues such as nonce errors, insufficient funds, etc. Adding 3 new endpoints for excluding panic txs.
 - `sei_traceBlockByNumberExcludeTraceFail`
 - `sei_getBlockByNumberExcludeTraceFail` + `sei_getBlockByHashExcludeTraceFail`
 -  `sei_getTransactionByReceiptExcludeTraceFail`

Relies on https://github.com/sei-protocol/go-ethereum/pull/40

## Testing performed to validate your change
added unit tests in rpc layer + test on RPC node

### CODE REVIEW

**Review by philipsu522 — APPROVED** [review]

File: evmrpc/block.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ import (
 )
 
 const ShellEVMTxType = math.MaxUint32
+const TestGenesisBlockHash = "0xF9D3845DF25B43B1C6926F3CEDA6845C17F5624E12212FD8847D0BA01DA1AB9E"
```
Comment (yzang2019) [inline evmrpc/block.go]: Why is this called TestGenesisBlockHash? Is this related to test at all?

**Review by yzang2019 — APPROVED** [review]

File: evmrpc/block.go (hunk 1 above)
Comment (jewei1997) [reply evmrpc/block.go]: ah good catch. Not really, moved it back to where it was before and got rid of the "Test" name.
